From d7ff7775e4b1fcfd5d86a88abf49467b97b5ed4b Mon Sep 17 00:00:00 2001
From: Sergey V. Udaltsov <svu@gnome.org>
Date: Mon, 11 Jan 2010 11:14:24 +0000
Subject: Extra parameter in xkl_engine_stop_listen

New libxklavier requires stop_listen function to specify exactly the
flags for stopping
---
diff --git a/configure.in b/configure.in
index af98caf..5540181 100644
--- a/configure.in
+++ b/configure.in
@@ -20,7 +20,7 @@ DBUS_GLIB_REQUIRED=0.34
 GCONF_REQUIRED=2.14.0
 GDK_REQUIRED=2.13
 GTK_REQUIRED=2.13
-LIBXKLAVIER_REQUIRED=4.0
+LIBXKLAVIER_REQUIRED=5.0
 
 AC_SUBST([DBUS_REQUIRED])
 AC_SUBST([DBUS_GLIB_REQUIRED])
diff --git a/libgnomekbd/gkbd-indicator.c b/libgnomekbd/gkbd-indicator.c
index 2ddd1f9..0c037a7 100644
--- a/libgnomekbd/gkbd-indicator.c
+++ b/libgnomekbd/gkbd-indicator.c
@@ -658,7 +658,8 @@ gkbd_indicator_start_listen (void)
 static void
 gkbd_indicator_stop_listen (void)
 {
-	xkl_engine_stop_listen (globals.engine);
+	xkl_engine_stop_listen (globals.engine,
+					XKLL_TRACK_KEYBOARD_STATE);
 
	gdk_window_remove_filter (NULL, (GdkFilterFunc)
						  gkbd_indicator_filter_x_evt, NULL);
--
cgit v0.8.3.1
